envi/archs/i386/emu.py

```diff
@@ -114,10 +115,68 @@ class BFastCall_Caller(BFastCall):
 msfastcall_caller = MsFastCall_Caller()
 bfastcall_caller = BFastCall_Caller()
 
+
+def doRepPrefix(emu, meth, op):
+    #FIXME check for opcode family valid to rep
+    ecx = emu.getRegister(REG_ECX)
+    if op.opcode == INS_NOP:
+        if exc == 0:
+            return
+        return meth(op)
+
+    ret = None
+    while ecx != 0:
+        ret = meth(op)
+        ecx -= 1
+        emu.setRegister(REG_ECX, ecx)
+    return ret
+
+def doRepnzPrefix(emu, meth, op):
+    #FIXME check for opcode family valid to rep
```

Comment:
We should probably just do this.

Reply:
i just removed it.  i'm not sure how to check it at this point.